addons/tmi/services/oauth.gd

```diff
@@ -250,8 +253,8 @@ func _code_to_token(code: String):
 		"image": payload.data.get("picture", "")
 	}
 	
-	newCredentials.channel = credentials.channel
-	newCredentials.broadcaster_user_id = await _lookup_channel(body.access_token)
+	newCredentials.channel = credentials.channel if !credentials.channel.is_empty() else payload.data.get("preferred_username", "")
```

Comment:
instead of extracting out preferred_username again, you can just use `newCredentials.user_login` here

Reply:
I can make that change - I'd been extracting it because capitalization was preserved in _idtoken_credentials.